htmresearch/algorithms/apical_tiebreak_bayesian_temporal_memory.py

```diff
@@ -450,8 +449,23 @@ def _updateMovingAverage(
       learningRate = self.learningRate
 
     numSegments = self.numApicalSegments if isApical else self.numBasalSegments
+
+    # Updating moving average input activity
+    noisy_input_vector = (1 - self.noise) * inputValues
+    # Consider only active segments
+    noisy_input_vector[noisy_input_vector > 0] += self.noise
+    movingAverageInput += learningRate * (
+            noisy_input_vector - movingAverageInput
+    )
+
+    # First update input values (includes learning rate)
+    # Then use the probabilities of activity for movingAverage calculation
+    # Instead of using 1 -> would lead to weight explosion, because we calculate weights based on MovingAverageInput
+    inputProbabilities = inputValues
+    inputProbabilities[inputValues.nonzero()] = movingAverageInput[inputValues.nonzero()]
+
     # Updating moving average weights to input
-    noisy_connection_matrix = np.outer((1 - self.noise**2) * segments, inputValues)
+    noisy_connection_matrix = np.outer((1 - self.noise**2) * segments, inputProbabilities)
```

Comment:
Good idea, but shouldn't the update only depend on the recent input / recent calculation? I think it is good for now but it might be worth to add a `TODO` comment or something to mark that. This might be a potential candidate for different experiment setups as well

Reply:
yeah thats the question, but it seems less odd than using the estimated activity for movingAverageInput and only 0/1 for movingAverage. In the experiments it lead to much later weight explosion/bias term.

Reply:
Hm, well, we can work with it for now